src/test/enhetstester/kotlin/no/nav/familie/ba/sak/kjerne/beregning/SmåbarnstilleggServiceTest.kt

```diff
@@ -0,0 +1,144 @@
+        småbarnstilleggService = SmåbarnstilleggService(
+            behandlingHentOgPersisterService = behandlingHentOgPersisterService,
+            efSakRestClient = efSakRestClient,
+            periodeOvergangsstønadGrunnlagRepository = periodeOvergangsstønadGrunnlagRepository,
+            tilkjentYtelseRepository = tilkjentYtelseRepository,
+            persongrunnlagService = persongrunnlagService,
+            andelerTilkjentYtelseOgEndreteUtbetalingerService = andelerTilkjentYtelseOgEndreteUtbetalingerService,
+            featureToggleService = featureToggleService,
+        )
+
+        every { periodeOvergangsstønadGrunnlagRepository.deleteByBehandlingId(any()) } just Runs
+        every { featureToggleService.isEnabled(FeatureToggleConfig.SATSENDRING_KOPIER_GRUNNLAG_FRA_FORRIGE_BEHANDLING) } returns true
+    }
+
+    @Test
+    fun `Ved satsendring skal gamle perioder kopieres`() {
+        val søker = lagPerson(type = PersonType.SØKER)
+        val fagsak = Fagsak(aktør = søker.aktør)
+        val forrigeBehandling = lagBehandling(årsak = BehandlingÅrsak.SØKNAD, fagsak = fagsak)
+        val behandling = lagBehandling(årsak = BehandlingÅrsak.SATSENDRING, fagsak = fagsak)
+
+        val perioderForrigeBehandling = listOf(
+            PeriodeOvergangsstønadGrunnlag(
+                fom = LocalDate.now().minusYears(1),
+                tom = LocalDate.now().minusMonths(8),
+                aktør = søker.aktør,
+                datakilde = Datakilde.EF,
+                behandlingId = forrigeBehandling.id,
+            ),
+            PeriodeOvergangsstønadGrunnlag(
+                fom = LocalDate.now().minusMonths(7),
+                tom = LocalDate.now().minusMonths(1),
+                aktør = søker.aktør,
+                datakilde = Datakilde.EF,
+                behandlingId = forrigeBehandling.id,
+            ),
+        )
+
+        val forventetNyePerioder = perioderForrigeBehandling.map {
+            PeriodeOvergangsstønadGrunnlag(
+                fom = it.fom,
+                tom = it.tom,
+                aktør = it.aktør,
+                datakilde = it.datakilde,
+                behandlingId = behandling.id,
+            )
+        }
+
+        every { behandlingHentOgPersisterService.hentForrigeBehandlingSomErVedtatt(behandling) } returns forrigeBehandling
+        every { periodeOvergangsstønadGrunnlagRepository.findByBehandlingId(forrigeBehandling.id) } returns perioderForrigeBehandling
+
+        val slot = slot<List<PeriodeOvergangsstønadGrunnlag>>()
+        every { periodeOvergangsstønadGrunnlagRepository.saveAll(capture(slot)) } returnsArgument 0
+
+        småbarnstilleggService.hentOgLagrePerioderMedOvergangsstønadForBehandling(
+            søkerAktør = søker.aktør,
+            behandling = behandling,
+        )
+
+        assertThat(slot.captured).containsAll(forventetNyePerioder)
+        verify(exactly = 1) { periodeOvergangsstønadGrunnlagRepository.saveAll(forventetNyePerioder) }
```

Comment:
Kan kanskje være fint å verifisere at man ikke har utført noe kall til ef-sak-klient her (selv om testen idag hadde feilet fordi den ikke er mocket)
```suggestion
        verify(exactly = 1) { periodeOvergangsstønadGrunnlagRepository.saveAll(forventetNyePerioder) }
        verify(exactly = 0) { efSakRestClient.hentPerioderMedFullOvergangsstønad(any()) }
```

Reply:
Kan eventuellt også verifisere at ikke-satsendring ikke henter forrige perioder